src/oscar/apps/offer/utils.py

```diff
@@ -96,7 +96,7 @@ def get_basket_offers(self, basket, user):
             return offers
 
         for voucher in basket.vouchers.all():
-            if voucher.is_active() and voucher.is_available_to_user(user):
```

Comment:
I'm not a fan of boolean variables like `no_message_on_result` - I would rather we used an index in the below conditional or read the two return values into variable - perhaps this:
```python
if voucher.is_active() and voucher.is_available_to_user(user=user)[0]:
```